Iterable/iterable-web-sdk
# Add Custom Close Button

## JIRA Ticket(s) if any

* [MOB-3869](https://iterable.atlassian.net/browse/MOB-3869)

## Description

Adds custom configurable close button to in-app messages. 

Read about it in the [README here](https://github.com/Iterable/iterable-web-sdk/blob/684952f1cf8fe1fa32be635dc494206c9de87ba3/README.md#usage)

API looks like this:

```ts
  const { request, pauseMessageStream, resumeMessageStream } = getInAppMessages(
    {
      count: 20,
      packageName: 'my-lil-website',
      closeButton: {
        color: 'white',
        position: 'top-right',
        size: 30,
        iconPath: './assets/somesvg.svg',
        sideOffset: '6%',
        topOffset: '6%'
      }
    },
    true
  );
```

## Screenshots

![Screen Shot 2022-02-10 at 4 17 34 PM](https://user-images.githubusercontent.com/7387001/153519034-0a05b915-60a8-460a-88d6-d06b22807c20.png)
![Screen Shot 2022-02-10 at 4 17 09 PM](https://user-images.githubusercontent.com/7387001/153519038-75d09f33-122f-4387-8749-3b4b2168f8d9.png)

## Test Steps

1. Run vanilla JS app with `yarn install:all && yarn start:all`
2. Send yourself an in-app message and paint it to the screen at http://localhost:8080
3. See custom close button in message

Additionally, play around with the other configuration values to make it appear differently. Supported values are:

```ts
  closeButton?: {
    position?: 'top-left' | 'top-right';
    color?: string;
    size?: string | number;
    iconPath?: string;
    topOffset?: string;
    sideOffset?: string;
  };
```
